### DISCUSSION TITLE
Bug in DTelegram

### DISCUSSION THREAD

**Discussion by 777petruh** [aso14/DTelegram#1] [Q&A]
Good afternoon. There are problems with your DTelegram plugin. An error occurs when you forward a message to yourself in a telegram bot.
![Безымянный](https://github.com/aso14/DTelegram/assets/18466470/86b0b389-a097-441a-bddf-c49a9695dfad)
![image](https://github.com/aso14/DTelegram/assets/18466470/88c3c173-73ea-4b95-a48d-33686c50d904)

**Comment by aso14** [comment]
Hi, which methods did you use to forward the message ?  Because the forward methods work using the  public channel Id. I didn't see it on your screenshot ! Please give me more information !!! 

`var Updates := TelegramBot1.GetUpdates(100, -1, 0);`

**Comment by 777petruh** [comment]
`{"ok":true,"result":[{"update_id":165418472,
"message":{"message_id":539,"from":{"id":1255784962,"is_bot":false,"first_name":"Simon","username":"HammeR_kos","language_code":"ru"},"chat":{"id":1255784962,"first_name":"Simon","username":"HammeR_kos","type":"private"},"date":1710380765,"text":"123"}},{"update_id":165418473,
"message":{"message_id":540,"from":{"id":1255784962,"is_bot":false,"first_name":"Simon","username":"HammeR_kos","language_code":"ru"},"chat":{"id":1255784962,"first_name":"Simon","username":"HammeR_kos","type":"private"},"date":1710380771,"reply_to_message":{"message_id":539,"from":{"id":1255784962,"is_bot":false,"first_name":"Simon","username":"HammeR_kos","language_code":"ru"},"chat":{"id":1255784962,"first_name":"Simon","username":"HammeR_kos","type":"private"},"date":1710380765,"text":"123"},"text":"123"}},{"update_id":165418474,
"message":{"message_id":541,"from":{"id":1255784962,"is_bot":false,"first_name":"Simon","username":"HammeR_kos","language_code":"ru"},"chat":{"id":1255784962,"first_name":"Simon","username":"HammeR_kos","type":"private"},"date":1710380827,"forward_origin":{"type":"user","sender_user":{"id":1255784962,"is_bot":false,"first_name":"Simon","username":"HammeR_kos","language_code":"ru"},"date":1710380771},"forward_from":{"id":1255784962,"is_bot":false,"first_name":"Simon","username":"HammeR_kos","language_code":"ru"},"forward_date":1710380771,"text":"123"}}]}`

error on "forward_origin"

I do it like this,
![123](https://github.com/aso14/DTelegram/assets/18466470/0fde8f81-74b8-4076-ae06-dea612c2e185)

I added a commit, I don't know if you see it or not.
If necessary, I can give lines of code on how I fixed everything.

**Reply by aso14** [reply]
I didn't see your commit ! Feel free to share the code with me !! I'll add it

**Reply by 777petruh** [reply]
Maybe I'm doing something wrong?
![image](https://github.com/aso14/DTelegram/assets/18466470/f4386aec-96ed-4bee-a92c-8c0254a2b37a)

**Reply by aso14** [reply]
Try it again  or use VS Code !!!
